src/protocol.rs

```diff
@@ -407,6 +416,132 @@ mod tests {
+            .iter()
+            .map(|config| Participant::<AuxInfoParticipant>::from_config(config, auxinfo_sid, ()))
+            .collect::<Vec<_>>();
+
+        let message_type = MessageType::Auxinfo(crate::messages::AuxinfoMessageType::Public);
+        let from = auxinfo_quorum[0].id;
+        let to = auxinfo_quorum[0].id;
+        let unverified_bytes = vec![0x01, 0x02, 0x03];
+        let message = Message::new(message_type, auxinfo_sid, from, to, &unverified_bytes);
+        let participant = auxinfo_quorum.iter_mut().choose(&mut rng).unwrap();
+        if let Err(e) = participant.process_single_message(&message, &mut rng) {
+            assert_eq!(
+                e,
+                InternalError::CallingApplicationMistake(CallerError::WrongMessageRecipient)
+            );
+        };
+        Ok(())
+    }
+
+    #[test]
+    fn process_single_message_wrong_protocol_type() -> Result<()> {
+        let mut rng = init_testing();
+        let QUORUM_SIZE = 3;
+        // Set GLOBAL config for participants
+        let configs = ParticipantConfig::random_quorum(QUORUM_SIZE, &mut rng);
+
+        // Set up auxinfo participants
+        let auxinfo_sid = Identifier::random(&mut rng);
+        let mut auxinfo_quorum = configs
+            .iter()
+            .map(|config| Participant::<AuxInfoParticipant>::from_config(config, auxinfo_sid, ()))
+            .collect::<Vec<_>>();
+
+        let message_type = MessageType::Keygen(crate::messages::KeygenMessageType::PublicKeyshare);
+        let from = auxinfo_quorum[0].id;
+        let to = auxinfo_quorum[0].id;
+        let unverified_bytes = vec![0x01, 0x02, 0x03];
+        let message = Message::new(message_type, auxinfo_sid, from, to, &unverified_bytes);
+        let participant = auxinfo_quorum.iter_mut().choose(&mut rng).unwrap();
+        if let Err(e) = participant.process_single_message(&message, &mut rng) {
+            assert_eq!(
+                e,
+                InternalError::CallingApplicationMistake(CallerError::WrongProtocol)
+            );
+        };
+        Ok(())
+    }
+
+    #[test]
+    fn process_single_message_wrong_sender_participant() -> Result<()> {
+        let mut rng = init_testing();
+        let mut rng1 = init_testing();
+        let QUORUM_SIZE = 3;
+        // Set GLOBAL config for participants
+        let configs = ParticipantConfig::random_quorum(QUORUM_SIZE, &mut rng);
+
+        // Set up auxinfo participants
+        let auxinfo_sid = Identifier::random(&mut rng);
+        let auxinfo_sid1 = Identifier::random(&mut rng1);
+        let mut auxinfo_quorum = configs
+            .iter()
+            .map(|config| Participant::<AuxInfoParticipant>::from_config(config, auxinfo_sid, ()))
+            .collect::<Vec<_>>();
+        let auxinfo_quorum1 = configs
+            .iter()
+            .map(|config| Participant::<AuxInfoParticipant>::from_config(config, auxinfo_sid1, ()))
+            .collect::<Vec<_>>();
```

Comment:
todo: Instead of generating a whole quorum here, you can create a single random participant ID, like
```rust
let bad_sender_pid = ParticipantIdentifier::random(&mut rng);
```